examples/camera/first_person_view_model.rs

```diff
@@ -0,0 +1,250 @@
+    let cube = meshes.add(Cuboid::new(2.0, 0.5, 1.0));
+    let material = materials.add(Color::WHITE);
+    // The world model camera will render the floor and the cube.
+    // Assigning no `RenderLayers` component defaults to layer 0.
+
+    commands.spawn(MaterialMeshBundle {
+        mesh: floor,
+        material: material.clone(),
+        ..default()
+    });
+
+    commands.spawn(MaterialMeshBundle {
+        mesh: cube.clone(),
+        material: material.clone(),
+        transform: Transform::from_xyz(0.0, 0.25, -3.0),
+        ..default()
+    });
+
+    commands.spawn(MaterialMeshBundle {
+        mesh: cube,
+        material,
+        transform: Transform::from_xyz(0.75, 1.75, 0.0),
+        ..default()
+    });
+}
+
+fn spawn_lights(mut commands: Commands) {
+    commands.spawn((
+        PointLightBundle {
+            point_light: PointLight {
+                color: Color::from(tailwind::ROSE_300),
+                shadows_enabled: true,
+                ..default()
+            },
+            transform: Transform::from_xyz(-2.0, 4.0, -0.75),
+            ..default()
+        },
+        // The light source illuminates both the view model and the world model.
+        RenderLayers::from_layers(&[0, 1]),
+    ));
+}
+
+fn spawn_text(mut commands: Commands) {
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                position_type: PositionType::Absolute,
+                bottom: Val::Px(12.0),
+                left: Val::Px(12.0),
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                concat!(
+                    "Move the camera with your mouse.\n",
+                    "Press arrow up to decrease the FOV of the world model.\n",
+                    "Press arrow down to increase the FOV of the world model."
+                ),
+                TextStyle {
+                    font_size: 25.0,
+                    ..default()
+                },
+            ));
+        });
+}
+
+// Functional systems.
+
+fn move_player(
+    mut mouse_motion: EventReader<MouseMotion>,
+    mut player: Query<&mut Transform, With<Player>>,
+) {
+    let mut transform = player.single_mut();
+    for motion in mouse_motion.read() {
+        let yaw = -motion.delta.x * 0.003;
+        let pitch = -motion.delta.y * 0.002;
+        // See <https://gamedev.stackexchange.com/a/136175/103059>
+        transform.rotate_y(yaw);
+        transform.rotate_local_x(pitch);
+    }
+}
+
+fn change_fov(
+    input: Res<ButtonInput<KeyCode>>,
+    mut world_model_projection: Query<&mut Projection, With<WorldModelCamera>>,
+) {
+    let mut projection = world_model_projection.single_mut();
+    let Projection::Perspective(ref mut perspective) = projection.as_mut() else {
+        // We've explicitly built this component with `Projection::Perspective`.
+        unreachable!();
```

Comment:
```suggestion
        unreachable!("component always built with `Projection::Perspective`");
```

Reply:
Oh, I didn't know that was valid! Adopting it with a slightly different wording.

Reply:
[`9291ef2` (#13828)](https://github.com/bevyengine/bevy/pull/13828/commits/9291ef2f7636a8a4a7d0fac948573d0e11277b55)